pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -163,6 +163,38 @@ func (a Adapter) Push(parameters common.PushParameters) (err error) {
 	return nil
 }
 
+// Test runs the devfile test command
+func (a Adapter) Test(testCmd string, show bool) (err error) {
+	// componentExists := utils.ComponentExists(a.Client, a.ComponentName)
+	// if !componentExists {
+	// 	return fmt.Errorf("component does not exist, a valid component is required to run 'odo test'")
+	// }
+	// pod, err := a.waitAndGetComponentPod(true)
+	// if err != nil {
+	// 	return errors.Wrapf(err, "unable to get pod for component %s", a.ComponentName)
+	// }
+
```

Comment:
Remember to update the integration test since its relying on the err msg

Reply:
The message`no Pod was found for the selector` in from `GetOnePodFromSelector` which is called by `GetPodUsingComponentName`. I prefer not update the existing code, since it was originally approved.

Reply:
I tried it does not wrap the err as expect. It because the error returned from  `GetPodUsingComponentName` is built using `fmt.Errorf`.

And I tried the following, it only shows `<some format string1>` in the terminal. It might be a general issue with our code & golang. 
 ```
errors.Wrapf(fmt.Errorf(<some format string1> ), <some format string2>)
```

Since the error message from `GetPodUsingComponentName` is clear enough, adding an additional error wrap is not meaningful; I will simply return err in this case.

Reply:
Ok, but it looks weird to me that the error is very generic. We should at least wrap it like`failed to run 'odo test': no Pod was found for the selector: component=xyz`

Reply:
IMO, after executing the command, the error message is shown with a red X, it's clear enough that `odo test` fail to run, so wrap wrap it like `failed to run 'odo test': no Pod was found for the selector: component=xyz` is meaningless. 

How about, `No valid Pod can be found to run 'odo test': no Pod was found for the selector: component=xyz`

Reply:
sounds good

Reply:
Or wrap as Mrinal suggest: `component is not running: <error message from GetPodUsingComponentName>`?
Which one do you think is more appropriate?

Reply:
`no valid pod can be found to run 'odo test'` since like you said the func that you're using can return 0 pod or more than 1 pod. So in the case of the latter, the component may be running but odo doesn't know which pod to use. So we should stick with what you suggested earlier